File: flowdesk/flowdesk_be/src/main/java/com/example/flowdesk_be/service/AuthService.java
package com.example.flowdesk_be.service;

import com.example.flowdesk_be.dto.request.LoginRequest;
import com.example.flowdesk_be.dto.request.RefreshTokenRequest;
import com.example.flowdesk_be.dto.request.RegisterRequest;
import com.example.flowdesk_be.dto.response.AuthResponse;

public interface AuthService {

  AuthResponse register(RegisterRequest request);

  AuthResponse login(LoginRequest request);

  // Dùng refresh token để lấy access token mới
  AuthResponse refresh(RefreshTokenRequest request);

  // Logout thiết bị hiện tại — revoke refresh token đang dùng
  void logout(RefreshTokenRequest request);

  // Logout tất cả thiết bị — revoke toàn bộ refresh token của user
  void logoutAll(String email);
}

File: flowdesk/flowdesk_fe/src/services/auth.service.ts
import { apiClient } from "@/lib/api";
import type {
  ApiResponse,
  AuthResponse,
  LoginRequest,
  RegisterRequest,
  RefreshTokenRequest,
} from "@/types";

export const authService = {
  login: async (payload: LoginRequest): Promise<AuthResponse> => {
    const { data } = await apiClient.post<ApiResponse<AuthResponse>>(
      "/api/auth/login",
      payload,
    );
    return data.data;
  },

  register: async (payload: RegisterRequest): Promise<AuthResponse> => {
    const { data } = await apiClient.post<ApiResponse<AuthResponse>>(
      "/api/auth/register",
      payload,
    );
    return data.data;
  },

  refresh: async (payload: RefreshTokenRequest): Promise<AuthResponse> => {
    const { data } = await apiClient.post<ApiResponse<AuthResponse>>(
      "/api/auth/refresh",
      payload,
    );
    return data.data;
  },

  logout: async (payload: RefreshTokenRequest): Promise<void> => {
    await apiClient.post("/api/auth/logout", payload);
  },

  logoutAll: async (): Promise<void> => {
    await apiClient.post("/api/auth/logout-all");
  },
};
